File: EmployeeService.java
package sg.nus.iss.team2.service;

import java.util.List;

import org.springframework.data.domain.Page;

import sg.nus.iss.team2.model.Employee;

public interface EmployeeService {

	List <Employee> getAllEmployees();      //getall method
	
	void saveEmployee(Employee employee);  //save method

	Employee getEmployeeById(long id);   //getbyid method
	
	void deleteEmployeeById(long id);   //delete method
	
	Page <Employee> findPaginated(int pageNo, int pageSize, String sortField, String sortDirection);   //using pagination API, must be sortDirection
}

File: src/Service/EmployeeService.js
import axios from "axios";

const EMPLOYEE_API_BASE_URL = "http://localhost:8080/api/v1/employee"

class EmployeeService {
    saveEmployee(employee){
        return axios.post(EMPLOYEE_API_BASE_URL,employee);
    }

    getEmployee(){
        return axios.get(EMPLOYEE_API_BASE_URL);
    }

    deleteEmployee(employeeId){
        return axios.delete(EMPLOYEE_API_BASE_URL+"/"+employeeId);
    }

    getEmployeeById(employeeId){
        return axios.get(EMPLOYEE_API_BASE_URL+"/"+employeeId);
    }

    updateEmployee(employee,employeeId){
        return axios.put(EMPLOYEE_API_BASE_URL+"/"+employeeId,employee);
    }

}

export default new EmployeeService();